crates/markdown_preview/src/markdown_preview_view.rs

```diff
@@ -0,0 +1,137 @@
+                return;
+            }
+            let languages = languages.clone();
+            if let Some(editor) = workspace.active_item_as::<Editor>(cx) {
+                let view: View<MarkdownPreviewView> =
+                    cx.new_view(|cx| MarkdownPreviewView::new(editor, languages, cx));
+                workspace.split_item(workspace::SplitDirection::Right, Box::new(view.clone()), cx);
+                cx.notify();
+            }
+        });
+    }
+
+    pub fn new(
+        active_editor: View<Editor>,
+        languages: Arc<LanguageRegistry>,
+        cx: &mut ViewContext<Self>,
+    ) -> Self {
+        let focus_handle = cx.focus_handle();
+
+        cx.subscribe(&active_editor, |this, editor, event: &EditorEvent, cx| {
+            if *event == EditorEvent::Edited {
+                let editor = editor.read(cx);
+                let contents = editor.buffer().read(cx).snapshot(cx).text();
+                this.contents = contents;
+                cx.notify();
+            }
+        })
+        .detach();
+
+        let editor = active_editor.read(cx);
+        let contents = editor.buffer().read(cx).snapshot(cx).text();
+
+        Self {
+            focus_handle,
+            languages,
+            contents,
+        }
+    }
+}
+
+impl FocusableView for MarkdownPreviewView {
+    fn focus_handle(&self, _: &AppContext) -> gpui::FocusHandle {
+        self.focus_handle.clone()
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum PreviewEvent {}
+
+impl EventEmitter<PreviewEvent> for MarkdownPreviewView {}
+
+impl Item for MarkdownPreviewView {
+    type Event = PreviewEvent;
+
+    fn tab_content(
+        &self,
+        _detail: Option<usize>,
+        selected: bool,
+        _cx: &WindowContext,
+    ) -> AnyElement {
+        h_flex()
+            .gap_2()
+            .child(Icon::new(IconName::FileDoc).color(if selected {
+                Color::Default
+            } else {
+                Color::Muted
+            }))
+            .child(Label::new("Markdown preview").color(if selected {
+                Color::Default
+            } else {
+                Color::Muted
+            }))
+            .into_any()
+    }
+
+    fn telemetry_event_text(&self) -> Option<&'static str> {
+        Some("markdown preview")
+    }
+
+    fn to_item_events(_event: &Self::Event, _f: impl FnMut(workspace::item::ItemEvent)) {}
+}
+
+impl Render for MarkdownPreviewView {
+    fn render(&mut self, cx: &mut ViewContext<Self>) -> impl IntoElement {
+        let mut container = v_flex()
+            .items_start()
+            .justify_start()
+            .key_context("MarkdownPreview")
+            .track_focus(&self.focus_handle)
+            .id("MarkdownPreview")
+            .overflow_scroll()
+            .size_full()
+            .bg(cx.theme().colors().editor_background)
+            .p_4();
+
+        for item in render_markdown(&self.contents, &self.languages, cx).into_iter() {
```

Comment:
I think we could replace this loop with a `.children` call, unless I'm missing something.

Reply:
I hadn't seen that yet. Handy!